### PR TITLE
Add support for add_torrent_file RPC call

### PR DESCRIPTION
**Pull request by prvn**

### CODE REVIEW

File: cli/cli.go [hunk 1]
```diff
@@ -132,6 +137,42 @@ func main() {
...
+		if err != nil {
+			fmt.Println("Error opening file :", err)
+			os.Exit(6)
+		}
+
+		defer file.Close()
+
+		fileName := filepath.Base(addFile)
+
+		data, err := ioutil.ReadAll(file)
```
Comment (gdm85) [inline cli/cli.go]: I suggest using `ioutil.ReadFile` instead

File: cli/cli.go [hunk 2]
```diff
@@ -132,6 +137,42 @@ func main() {
...
+		if err != nil {
+			fmt.Println("Error reading file :", err)
+			os.Exit(6)
+		}
+
+		var buffer bytes.Buffer
+		encoder := base64.NewEncoder(base64.StdEncoding, &buffer)
```
Comment (gdm85) [inline cli/cli.go]: I suggest using `base64.StdEncoding.EncodeToString()` here

File: delugeclient.go [hunk 3]
```diff
@@ -46,6 +46,7 @@ type DelugeClient interface {
 	DaemonVersion() (string, error)
 	GetFreeSpace(string) (int64, error)
 	AddTorrentMagnet(magnetURI string, options Options) (string, error)
+	AddTorrentFile(fileName, fileDump string, options Options) (string, error)
```
Comment (gdm85) [inline delugeclient.go]: It would be better if `fileDump` were instead `fileContentBase64`

**Comment by prvn** [comment]
Updated code and squashed the commits

File: cli/cli.go [hunk 4]
```diff
@@ -16,16 +16,18 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
...
 	"fmt"
 	"io/ioutil"
 	"log"
 	"os"
+	"path/filepath"
 	"strings"
 	"time"
 
-	"github.com/gdm85/go-libdeluge"
+	"github.com/prvn/go-libdeluge"
```
Comment (gdm85) [inline cli/cli.go]: @prvn can you please fix this?

File: go.mod [hunk 5]
```diff
@@ -1,4 +1,4 @@
-module github.com/gdm85/go-libdeluge
+module github.com/prvn/go-libdeluge
```
Comment (gdm85) [inline go.mod]: Same here.

**Comment by prvn** [comment]
I think I messed up this PR with local changes, will open a new one

**Pull request closed without merging**